// @dhis2-ui/organisation-unit-tree/build/cjs/organisation-unit-tree/use-root-org-data/use-root-org-data.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.useRootOrgData = exports.createRootQuery = void 0;

var _appRuntime = require("@dhis2/app-runtime");

var _react = require("react");

var _patchMissingDisplayName = require("./patch-missing-display-name.js");

const createRootQuery = ids => ids.reduce((query, id) => ({ ...query,
  [id]: {
    id,
    resource: `organisationUnits`,
    params: ({
      isUserDataViewFallback
    }) => ({
      isUserDataViewFallback,
      fields: ['displayName', 'path', 'id'],
      paging: false
    })
  }
}), {});
/**
 * @param {string[]} ids
 * @param {Object} [options]
 * @param {boolean} [options.withChildren]
 * @param {boolean} [options.isUserDataViewFallback]
 * @returns {Object}
 */


exports.createRootQuery = createRootQuery;

const useRootOrgData = (ids, {
  isUserDataViewFallback
} = {}) => {
  const query = createRootQuery(ids);
  const variables = {
    isUserDataViewFallback
  };
  const {
    loading,
    error,
    data,
    refetch
  } = (0, _appRuntime.useDataQuery)(query, {
    variables
  });
  const patchedData = (0, _react.useMemo)(() => {
    return data ? (0, _patchMissingDisplayName.patchMissingDisplayName)(data) : data;
  }, [data]);
  return {
    loading,
    error: error || null,
    data: patchedData || null,
    refetch
  };
};

exports.useRootOrgData = useRootOrgData;
