// @dhis2-ui/sharing-dialog/build/es/access-select.js
import _JSXStyle from "styled-jsx/style";
import { MenuDivider, MenuItem } from '@dhis2-ui/menu';
import { SingleSelectField } from '@dhis2-ui/select';
import { Tooltip } from '@dhis2-ui/tooltip';
import { useOnlineStatus } from '@dhis2/app-runtime';
import PropTypes from '@dhis2/prop-types';
import React from 'react';
import i18n from './locales/index.js';
import { accessStrings } from './sharing-constants.js';
import { accessSelectStyles } from './sharing-dialog.styles.js';
export const AccessSelect = ({
  label,
  placeholder,
  prefix,
  access,
  accessOptions,
  showRemoveOption,
  disabled,
  onChange
}) => {
  const {
    offline
  } = useOnlineStatus();

  const renderSingleSelectField = () => /*#__PURE__*/React.createElement(SingleSelectField, {
    label: label,
    placeholder: placeholder,
    prefix: prefix,
    disabled: offline || disabled,
    selected: access,
    onChange: ({
      selected
    }) => onChange(selected)
  }, accessOptions.map(value => /*#__PURE__*/React.createElement(CustomSingleSelectOption, {
    key: value,
    label: accessStrings[value].option,
    value: value,
    active: value === access
  })), showRemoveOption && /*#__PURE__*/React.createElement(CustomSingleSelectOption, {
    key: "remove",
    label: i18n.t('Remove access'),
    value: "remove",
    destructive: true
  }));

  return /*#__PURE__*/React.createElement("div", {
    className: `jsx-${accessSelectStyles.__hash}` + " " + "share-select"
  }, /*#__PURE__*/React.createElement(_JSXStyle, {
    id: accessSelectStyles.__hash
  }, accessSelectStyles), offline ? /*#__PURE__*/React.createElement(Tooltip, {
    content: i18n.t('Not available offline')
  }, renderSingleSelectField()) : renderSingleSelectField());
};
AccessSelect.propTypes = {
  accessOptions: PropTypes.array.isRequired,
  onChange: PropTypes.func.isRequired,
  access: PropTypes.string,
  disabled: PropTypes.bool,
  label: PropTypes.string,
  placeholder: PropTypes.string,
  prefix: PropTypes.string,
  showRemoveOption: PropTypes.bool
};

const CustomSingleSelectOption = ({
  label,
  value,
  active,
  destructive,
  onClick
}) => /*#__PURE__*/React.createElement(React.Fragment, null, destructive && /*#__PURE__*/React.createElement(MenuDivider, {
  dense: true
}), /*#__PURE__*/React.createElement(MenuItem, {
  label: label,
  value: value,
  active: active,
  destructive: destructive,
  onClick: (_, e) => onClick({
    selected: value
  }, e),
  dense: true
}));

CustomSingleSelectOption.propTypes = {
  label: PropTypes.string.isRequired,
  value: PropTypes.string.isRequired,
  active: PropTypes.bool,
  destructive: PropTypes.bool,
  onClick: PropTypes.func
};
