// @dhis2-ui/pagination/build/es/page-summary.js
import _JSXStyle from "styled-jsx/style";
import propTypes from '@dhis2/prop-types';
import { colors, spacers } from '@dhis2/ui-constants';
import React from 'react'; // TODO: i18n translate

const translate = (prop, interpolationObject) => {
  if (typeof prop === 'function') {
    return prop(interpolationObject);
  }

  return prop;
};

const getItemRange = (page, pageSize, total) => {
  let firstItem, lastItem;

  if (total === 0) {
    // if no items are found, the pager total is 0
    // and we want to force the first and last item to be 0 too
    firstItem = 0;
    lastItem = 0;
  } else {
    // page is 1-based
    firstItem = (page - 1) * pageSize + 1;
    lastItem = firstItem + pageSize - 1;
  }

  if (lastItem > total) {
    lastItem = total;
  }

  return {
    firstItem,
    lastItem
  };
};

const PageSummary = ({
  dataTest,
  page,
  pageCount,
  pageSize,
  pageSummaryText,
  total
}) => {
  const {
    firstItem,
    lastItem
  } = getItemRange(page, pageSize, total);
  const summary = translate(pageSummaryText, {
    page,
    pageCount,
    firstItem,
    lastItem,
    total
  });
  return /*#__PURE__*/React.createElement("div", {
    "data-test": `${dataTest}-summary`,
    className: _JSXStyle.dynamic([["3931741314", [spacers.dp12, colors.grey700]]])
  }, /*#__PURE__*/React.createElement("span", {
    className: _JSXStyle.dynamic([["3931741314", [spacers.dp12, colors.grey700]]])
  }, summary), /*#__PURE__*/React.createElement(_JSXStyle, {
    id: "3931741314",
    dynamic: [spacers.dp12, colors.grey700]
  }, [`div.__jsx-style-dynamic-selector{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-flex-shrink:0;-ms-flex-negative:0;flex-shrink:0;min-height:32px;margin-right:${spacers.dp12};}`, `span.__jsx-style-dynamic-selector{color:${colors.grey700};font-size:14px;}`]));
};

PageSummary.propTypes = {
  dataTest: propTypes.string.isRequired,
  page: propTypes.number.isRequired,
  pageCount: propTypes.number.isRequired,
  pageSize: propTypes.number.isRequired,
  pageSummaryText: propTypes.oneOfType([propTypes.string, propTypes.func]).isRequired,
  total: propTypes.number.isRequired
};
export { PageSummary, getItemRange };
