// @dhis2-ui/pagination/build/es/page-size-select.js
import _JSXStyle from "styled-jsx/style";
import { SingleSelect, SingleSelectOption } from '@dhis2-ui/select';
import propTypes from '@dhis2/prop-types';
import { spacers } from '@dhis2/ui-constants';
import React from 'react'; // TODO: i18n translate

const translate = (prop, interpolationObject) => {
  if (typeof prop === 'function') {
    return prop(interpolationObject);
  }

  return prop;
};

const PageSizeSelect = ({
  dataTest,
  pageSizeSelectText,
  pageSize,
  pageSizes,
  onChange
}) => /*#__PURE__*/React.createElement("div", {
  "data-test": `${dataTest}-pagesize`,
  className: _JSXStyle.dynamic([["1934482915", [spacers.dp12]]])
}, /*#__PURE__*/React.createElement(SingleSelect, {
  dense: true,
  selected: pageSize.toString(),
  onChange: ({
    selected
  }) => onChange(parseInt(selected, 10)),
  className: "select",
  dataTest: `${dataTest}-pagesize-select`,
  prefix: translate(pageSizeSelectText)
}, pageSizes.map(length => /*#__PURE__*/React.createElement(SingleSelectOption, {
  key: length,
  value: length,
  label: length
}))), /*#__PURE__*/React.createElement(_JSXStyle, {
  id: "1934482915",
  dynamic: [spacers.dp12]
}, [`div.__jsx-style-dynamic-selector{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-flex-shrink:0;-ms-flex-negative:0;flex-shrink:0;min-height:32px;margin-right:${spacers.dp12};-webkit-box-flex:1;-webkit-flex-grow:1;-ms-flex-positive:1;flex-grow:1;}`]));

PageSizeSelect.propTypes = {
  dataTest: propTypes.string.isRequired,
  pageSize: propTypes.number.isRequired,
  pageSizeSelectText: propTypes.oneOfType([propTypes.string, propTypes.func]).isRequired,
  pageSizes: propTypes.arrayOf(propTypes.string).isRequired,
  onChange: propTypes.func.isRequired
};
export { PageSizeSelect };
