// @dhis2-ui/sharing-dialog/build/es/autocomplete/menu-wrapper.js
import _JSXStyle from "styled-jsx/style";
import { Card } from '@dhis2-ui/card';
import { Layer } from '@dhis2-ui/layer';
import { Popper } from '@dhis2-ui/popper';
import PropTypes from '@dhis2/prop-types';
import React from 'react';

const MenuWrapper = ({
  children,
  dataTest,
  maxHeight,
  menuWidth,
  onClick,
  menuRef
}) => {
  const {
    styles,
    className: cardClassName
  } = {
    styles: /*#__PURE__*/React.createElement(_JSXStyle, {
      id: "81273804",
      dynamic: [maxHeight]
    }, [`.__jsx-style-dynamic-selector{height:auto;max-height:${maxHeight};overflow:auto;}`]),
    className: _JSXStyle.dynamic([["81273804", [maxHeight]]])
  };
  return /*#__PURE__*/React.createElement(Layer, {
    onClick: onClick,
    transparent: true
  }, /*#__PURE__*/React.createElement(Popper, {
    reference: menuRef,
    placement: "bottom",
    observeReferenceResize: true
  }, /*#__PURE__*/React.createElement("div", {
    "data-test": `${dataTest}-menuwrapper`,
    className: _JSXStyle.dynamic([["3936600096", [menuWidth]]])
  }, /*#__PURE__*/React.createElement(Card, {
    className: cardClassName
  }, children), styles, /*#__PURE__*/React.createElement(_JSXStyle, {
    id: "3936600096",
    dynamic: [menuWidth]
  }, [`div.__jsx-style-dynamic-selector{width:${menuWidth};}`]))));
};

MenuWrapper.defaultProps = {
  maxHeight: '280px'
};
MenuWrapper.propTypes = {
  dataTest: PropTypes.string.isRequired,
  menuRef: PropTypes.object.isRequired,
  menuWidth: PropTypes.string.isRequired,
  children: PropTypes.node,
  maxHeight: PropTypes.string,
  onClick: PropTypes.func
};
export { MenuWrapper };
