// @dhis2-ui/sharing-dialog/build/cjs/dashboard-cascade-sharing-content.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.DashboardCascadeSharingContent = void 0;

var _style = _interopRequireDefault(require("styled-jsx/style"));

var _button = require("@dhis2-ui/button");

var _loader = require("@dhis2-ui/loader");

var _noticeBox = require("@dhis2-ui/notice-box");

var _tooltip = require("@dhis2-ui/tooltip");

var _appRuntime = require("@dhis2/app-runtime");

var _propTypes = _interopRequireDefault(require("@dhis2/prop-types"));

var _uiConstants = require("@dhis2/ui-constants");

var _uiIcons = require("@dhis2/ui-icons");

var _react = _interopRequireWildcard(require("react"));

var _index = _interopRequireDefault(require("./locales/index.js"));

var _sharingDialogStyles = require("./sharing-dialog.styles.js");

function _getRequireWildcardCache(nodeInterop) { if (typeof WeakMap !== "function") return null; var cacheBabelInterop = new WeakMap(); var cacheNodeInterop = new WeakMap(); return (_getRequireWildcardCache = function (nodeInterop) { return nodeInterop ? cacheNodeInterop : cacheBabelInterop; })(nodeInterop); }

function _interopRequireWildcard(obj, nodeInterop) { if (!nodeInterop && obj && obj.__esModule) { return obj; } if (obj === null || typeof obj !== "object" && typeof obj !== "function") { return { default: obj }; } var cache = _getRequireWildcardCache(nodeInterop); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (key !== "default" && Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj.default = obj; if (cache) { cache.set(obj, newObj); } return newObj; }

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

const dashboardQuery = {
  dashboard: {
    resource: 'dashboards',
    id: ({
      id
    }) => id,
    params: {
      fields: 'dashboardItems[type]'
    }
  }
};

const getVisualizationsCount = queryResponse => {
  var _queryResponse$dashbo;

  return (queryResponse === null || queryResponse === void 0 ? void 0 : (_queryResponse$dashbo = queryResponse.dashboard.dashboardItems) === null || _queryResponse$dashbo === void 0 ? void 0 : _queryResponse$dashbo.length) > 0 ? queryResponse.dashboard.dashboardItems.filter(item => ['VISUALIZATION', 'MAP', 'EVENT_CHART', 'EVENT_REPORT'].includes(item.type)).length : 0;
};

const getCascadeSharingMutation = id => ({
  resource: `dashboards/cascadeSharing/${id}`,
  type: 'create'
});

const DashboardCascadeSharingContent = ({
  sharingSettings
}) => {
  const {
    offline
  } = (0, _appRuntime.useOnlineStatus)();
  const {
    data: queryResponse
  } = (0, _appRuntime.useDataQuery)(dashboardQuery, {
    variables: {
      id: sharingSettings.id
    }
  });
  const visualizationsCount = getVisualizationsCount(queryResponse);
  const mutation = (0, _react.useMemo)(() => getCascadeSharingMutation(sharingSettings.id), []);
  const [mutate, {
    loading,
    error,
    data: mutationResponse
  }] = (0, _appRuntime.useDataMutation)(mutation);
  const usersGroupsCount = Object.keys(sharingSettings.users).length + Object.keys(sharingSettings.groups).length;

  const renderApplySharingButton = () => /*#__PURE__*/_react.default.createElement(_button.Button, {
    type: "button",
    disabled: offline || loading || !usersGroupsCount,
    secondary: true,
    onClick: () => mutate()
  }, _index.default.t('Apply sharing to dashboard visualizations'));

  const getInfoMessage = () => {
    let message;

    if (usersGroupsCount > 0) {
      const messagePart1 = _index.default.t('{{count}} visualization on this dashboard will potentially get updated sharing settings.', {
        count: visualizationsCount,
        defaultValue: '{{count}} visualization on this dashboard will potentially get updated sharing settings.',
        defaultValue_plural: 'All {{count}} visualizations on this dashboard will potentially get updated sharing settings.'
      });

      message = messagePart1 + ' ' + _index.default.t('These updated sharing settings will apply to {{count}} user or group.', {
        count: usersGroupsCount,
        defaultValue: 'These updated sharing settings will apply to {{count}} user or group.',
        defaultValue_plural: 'These updated sharing settings will apply to {{count}} users or groups.'
      });
    } else {
      message = _index.default.t('There are no users, groups or roles to apply sharing settings for.');
    }

    return /*#__PURE__*/_react.default.createElement(_react.default.Fragment, null, /*#__PURE__*/_react.default.createElement(_style.default, {
      id: _sharingDialogStyles.cascadeSharingStyles.__hash
    }, _sharingDialogStyles.cascadeSharingStyles), /*#__PURE__*/_react.default.createElement("div", {
      className: `jsx-${_sharingDialogStyles.cascadeSharingStyles.__hash}` + " " + "box-block"
    }, /*#__PURE__*/_react.default.createElement("div", {
      className: `jsx-${_sharingDialogStyles.cascadeSharingStyles.__hash}` + " " + "box box-info"
    }, /*#__PURE__*/_react.default.createElement(_uiIcons.IconInfo16, {
      color: _uiConstants.colors.grey900
    }), /*#__PURE__*/_react.default.createElement("span", {
      className: `jsx-${_sharingDialogStyles.cascadeSharingStyles.__hash}` + " " + "box-text"
    }, message))));
  };

  const getResultMessage = () => {
    let message;

    if ((mutationResponse === null || mutationResponse === void 0 ? void 0 : mutationResponse.errorReports.length) === 0) {
      const updatedItems = mutationResponse.countUpdatedDashboardItems;

      if (updatedItems === 0) {
        message = _index.default.t('No visualizations were updated because sharing settings are already sufficient.');
        return /*#__PURE__*/_react.default.createElement(_noticeBox.NoticeBox, null, message);
      } else if (updatedItems === visualizationsCount) {
        message = _index.default.t('Successfully updated sharing for {{count}} visualization.', {
          count: visualizationsCount,
          defaultValue: 'Successfully updated sharing for {{count}} visualization.',
          defaultValue_plural: 'Successfully updated sharing for {{count}} visualizations.'
        });
      } else if (updatedItems < visualizationsCount) {
        // split in 2 because of nesting with plurals not working
        message = _index.default.t('{{count}} visualization was updated.', {
          count: updatedItems,
          defaultValue: '{{count}} visualization was updated.',
          defaultValue_plural: '{{count}} visualizations were updated.'
        }) + ' ' + _index.default.t('{{count}} visualization already has sufficient sharing settings.', {
          count: visualizationsCount - updatedItems,
          defaultValue: '{{count}} visualization already has sufficient sharing settings.',
          defaultValue_plural: '{{count}} visualizations already have sufficient sharing settings.'
        });
      }

      return /*#__PURE__*/_react.default.createElement(_react.default.Fragment, null, /*#__PURE__*/_react.default.createElement(_style.default, {
        id: _sharingDialogStyles.cascadeSharingStyles.__hash
      }, _sharingDialogStyles.cascadeSharingStyles), /*#__PURE__*/_react.default.createElement("div", {
        className: `jsx-${_sharingDialogStyles.cascadeSharingStyles.__hash}` + " " + "box-block"
      }, /*#__PURE__*/_react.default.createElement("div", {
        className: `jsx-${_sharingDialogStyles.cascadeSharingStyles.__hash}` + " " + "box box-success"
      }, /*#__PURE__*/_react.default.createElement(_uiIcons.IconCheckmark16, {
        color: _uiConstants.colors.grey700
      }), /*#__PURE__*/_react.default.createElement("span", {
        className: `jsx-${_sharingDialogStyles.cascadeSharingStyles.__hash}` + " " + "box-text"
      }, message))));
    } else {
      const messageCheckPermissions = _index.default.t('Check that you have permission to change sharing for all visualizations.');

      if ((mutationResponse === null || mutationResponse === void 0 ? void 0 : mutationResponse.countUpdatedDashboardItems) === 0) {
        message = _index.default.t('No visualizations were updated.') + ' ' + messageCheckPermissions;
      } else if (mutationResponse !== null && mutationResponse !== void 0 && mutationResponse.countUpdatedDashboardItems && queryResponse !== null && queryResponse !== void 0 && queryResponse.dashboard.itemsCount) {
        const messagePart1 = _index.default.t('{{count}} visualization was updated', {
          count: mutationResponse.countUpdatedDashboardItems,
          defaultValue: '{{count}} visualization was updated',
          defaultValue_plural: '{{count}} visualizations were updated.'
        });

        message = messagePart1 + ' ' + _index.default.t('{{count}} visualization could not be updated.', {
          count: visualizationsCount - mutationResponse.countUpdatedDashboardItems,
          defaultValue: '{{count}} visualization could not be updated.',
          defaultValue_plural: '{{count}} visualizations could not be updated.'
        }) + ' ' + messageCheckPermissions;
      }

      return /*#__PURE__*/_react.default.createElement(_noticeBox.NoticeBox, {
        warning: true
      }, message);
    }
  };

  return /*#__PURE__*/_react.default.createElement(_react.default.Fragment, null, /*#__PURE__*/_react.default.createElement(_style.default, {
    id: _sharingDialogStyles.cascadeSharingStyles.__hash
  }, _sharingDialogStyles.cascadeSharingStyles), /*#__PURE__*/_react.default.createElement("div", {
    className: `jsx-${_sharingDialogStyles.cascadeSharingStyles.__hash}` + " " + "title"
  }, _index.default.t('Apply dashboard sharing settings to visualizations')), /*#__PURE__*/_react.default.createElement("div", {
    className: `jsx-${_sharingDialogStyles.cascadeSharingStyles.__hash}` + " " + "description"
  }, _index.default.t('Applying the same sharing settings makes sure that users, groups and roles that have access to the dashboard also have at least "View only" access to its visualizations (charts, tables, maps, event charts, event reports).')), /*#__PURE__*/_react.default.createElement("div", {
    className: `jsx-${_sharingDialogStyles.cascadeSharingStyles.__hash}` + " " + "description"
  }, _index.default.t('If a user, group, or role already has "View and edit" access to a visualization, this won\'t be reduced to "View only". "All users" access level won\'t be updated or changed.')), /*#__PURE__*/_react.default.createElement("div", {
    className: `jsx-${_sharingDialogStyles.cascadeSharingStyles.__hash}` + " " + "description"
  }, _index.default.t("Applying sharing can't be undone, and needs to done again if new visualizations are added to the dashboard or its sharing settings are changed.")), visualizationsCount && getInfoMessage(), offline ? /*#__PURE__*/_react.default.createElement(_tooltip.Tooltip, {
    content: _index.default.t('Not available offline')
  }, renderApplySharingButton()) : renderApplySharingButton(), loading && /*#__PURE__*/_react.default.createElement("div", {
    className: `jsx-${_sharingDialogStyles.cascadeSharingStyles.__hash}` + " " + "loading"
  }, /*#__PURE__*/_react.default.createElement(_loader.CircularLoader, {
    small: true
  }), ' ', /*#__PURE__*/_react.default.createElement("span", {
    className: `jsx-${_sharingDialogStyles.cascadeSharingStyles.__hash}`
  }, _index.default.t('Updating sharing settings...'))), /*#__PURE__*/_react.default.createElement("div", {
    className: `jsx-${_sharingDialogStyles.cascadeSharingStyles.__hash}` + " " + "result-box"
  }, error && /*#__PURE__*/_react.default.createElement(_noticeBox.NoticeBox, {
    error: true
  }, _index.default.t('There was a problem updating dashboard visualizations. No visualizations were updated. Try again, or contact a system administrator.')), mutationResponse && getResultMessage()));
};

exports.DashboardCascadeSharingContent = DashboardCascadeSharingContent;
DashboardCascadeSharingContent.propTypes = {
  sharingSettings: _propTypes.default.object.isRequired
};
