// @dhis2-ui/organisation-unit-tree/build/cjs/organisation-unit-node/use-org-data/use-org-data.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.useOrgData = void 0;

var _appRuntime = require("@dhis2/app-runtime");

var _react = require("react");

var _index = require("../../helpers/index.js");

const ORG_DATA_QUERY = {
  orgUnit: {
    resource: `organisationUnits`,
    id: ({
      id
    }) => id,
    params: ({
      isUserDataViewFallback
    }) => ({
      isUserDataViewFallback,
      fields: ['id', 'path', 'children[id,path,displayName]']
    })
  }
};
/**
 * @param {string[]} ids
 * @param {Object} options
 * @param {string} options.displayName
 * @param {boolean} [options.withChildren]
 * @returns {Object}
 */

const useOrgData = (id, {
  isUserDataViewFallback,
  suppressAlphabeticalSorting,
  onComplete,
  displayName
}) => {
  if (!displayName) {
    throw new Error('"displayName" is required');
  }

  const defaultData = {
    id,
    displayName
  };
  const variables = {
    id,
    isUserDataViewFallback
  };
  const [state, setState] = (0, _react.useState)({
    loading: true,
    error: null,
    data: defaultData
  });
  (0, _appRuntime.useDataQuery)(ORG_DATA_QUERY, {
    variables,
    onComplete: ({
      orgUnit: node
    }) => {
      const sorted = !suppressAlphabeticalSorting ? (0, _index.sortNodeChildrenAlphabetically)(node) : node;
      const nextData = { ...defaultData,
        ...sorted
      };
      setState({ ...state,
        data: nextData,
        loading: false
      });
      onComplete && onComplete(nextData);
    },
    onError: queryError => {
      setState({ ...state,
        error: queryError,
        loading: false
      });
    }
  });
  return state;
};

exports.useOrgData = useOrgData;
