// @dhis2-ui/header-bar/build/cjs/header-bar.stories.e2e.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.WithLastOnline = exports.PWAEnabled = exports.ShowOnlineStatus = exports.Default = exports.default = void 0;

var _appRuntime = require("@dhis2/app-runtime");

var _react = _interopRequireDefault(require("react"));

var _headerBar = require("./header-bar.js");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var _default = {
  title: 'HeaderBarTesting',
  component: _headerBar.HeaderBar
};
exports.default = _default;
const config = {
  baseUrl: 'https://domain.tld/',
  apiVersion: ''
};

const Default = () => /*#__PURE__*/_react.default.createElement(_appRuntime.Provider, {
  config: config
}, /*#__PURE__*/_react.default.createElement(_headerBar.HeaderBar, {
  appName: "Example!"
}));

exports.Default = Default;

const ShowOnlineStatus = () => /*#__PURE__*/_react.default.createElement(_appRuntime.Provider, {
  config: { ...config,
    headerbar: {
      showOnlineStatus: true
    }
  }
}, /*#__PURE__*/_react.default.createElement(_headerBar.HeaderBar, {
  appName: "Example!"
}));

exports.ShowOnlineStatus = ShowOnlineStatus;

const PWAEnabled = () => /*#__PURE__*/_react.default.createElement(_appRuntime.Provider, {
  config: { ...config,
    pwaEnabled: true
  }
}, /*#__PURE__*/_react.default.createElement(_headerBar.HeaderBar, {
  appName: "Example!"
}));

exports.PWAEnabled = PWAEnabled;

const WithLastOnline = () => /*#__PURE__*/_react.default.createElement(_appRuntime.Provider, {
  config: { ...config,
    headerbar: {
      showOnlineStatus: true,
      onlineStatusInfo: 'LAST_ONLINE'
    }
  }
}, /*#__PURE__*/_react.default.createElement(_headerBar.HeaderBar, {
  appName: "Example!"
}));

exports.WithLastOnline = WithLastOnline;
