// react-docgen/dist/config.js
import { childContextTypeHandler, codeTypeHandler, componentDocblockHandler, componentMethodsHandler, componentMethodsJsDocHandler, contextTypeHandler, defaultPropsHandler, displayNameHandler, propDocBlockHandler, propTypeCompositionHandler, propTypeHandler, } from './handlers/index.js';
import { fsImporter } from './importer/index.js';
import { findExportedComponentDefinition } from './resolver/index.js';
const defaultResolver = findExportedComponentDefinition;
const defaultImporter = fsImporter;
export const defaultHandlers = [
    propTypeHandler,
    contextTypeHandler,
    childContextTypeHandler,
    propTypeCompositionHandler,
    propDocBlockHandler,
    codeTypeHandler,
    defaultPropsHandler,
    componentDocblockHandler,
    displayNameHandler,
    componentMethodsHandler,
    componentMethodsJsDocHandler,
];
export function createConfig(inputConfig) {
    const { babelOptions, filename, handlers, importer, resolver } = inputConfig;
    const config = {
        babelOptions: { ...babelOptions },
        handlers: handlers ?? defaultHandlers,
        importer: importer ?? defaultImporter,
        resolver: resolver ?? defaultResolver,
    };
    if (filename) {
        config.babelOptions.filename = filename;
    }
    return config;
}
