// @dhis2-ui/transfer/build/es/filter.js
import _JSXStyle from "styled-jsx/style";
import { Field } from '@dhis2-ui/field';
import { Input } from '@dhis2-ui/input';
import propTypes from '@dhis2/prop-types';
import { spacers } from '@dhis2/ui-constants';
import React from 'react';
export const Filter = ({
  dataTest,
  filter,
  onChange,
  label,
  placeholder
}) => /*#__PURE__*/React.createElement("div", {
  "data-test": dataTest,
  className: _JSXStyle.dynamic([["3679907342", [spacers.dp8, spacers.dp8]]])
}, /*#__PURE__*/React.createElement(Field, {
  label: label,
  name: dataTest,
  dataTest: `${dataTest}-field`
}, /*#__PURE__*/React.createElement(Input, {
  dataTest: `${dataTest}-input`,
  type: "search",
  placeholder: placeholder,
  name: dataTest,
  value: filter,
  onChange: onChange
})), /*#__PURE__*/React.createElement(_JSXStyle, {
  id: "3679907342",
  dynamic: [spacers.dp8, spacers.dp8]
}, [`div.__jsx-style-dynamic-selector{padding-bottom:${spacers.dp8};}`, `div.__jsx-style-dynamic-selector:first-child{padding-top:${spacers.dp8};}`]));
Filter.propTypes = {
  dataTest: propTypes.string.isRequired,
  filter: propTypes.string.isRequired,
  onChange: propTypes.func.isRequired,
  label: propTypes.string,
  placeholder: propTypes.string
};
