// @dhis2-ui/transfer/build/es/right-side.js
import _JSXStyle from "styled-jsx/style";
import propTypes from '@dhis2/prop-types';
import React from 'react';
import { backgroundColor, borderColor, borderRadius } from './common/index.js';
export const RightSide = ({
  children,
  dataTest,
  width
}) => /*#__PURE__*/React.createElement("div", {
  "data-test": dataTest,
  className: _JSXStyle.dynamic([["1391448013", [backgroundColor, borderColor, borderRadius, width]]])
}, children,
/**
 * Flex basis 0px to make sure right and left side
 * always have the same width
 */
'', /*#__PURE__*/React.createElement(_JSXStyle, {
  id: "1391448013",
  dynamic: [backgroundColor, borderColor, borderRadius, width]
}, [`div.__jsx-style-dynamic-selector{background-color:${backgroundColor};border:1px solid ${borderColor};border-radius:${borderRadius};display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-flex-direction:column;-ms-flex-direction:column;flex-direction:column;max-width:100%;width:${width};}`]));
RightSide.propTypes = {
  width: propTypes.string.isRequired,
  children: propTypes.node,
  dataTest: propTypes.string
};
