// @dhis2-ui/transfer/build/cjs/__e2e__/add_remove-highlighted-options.stories.e2e.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.HasSelected = exports.HasOptions = exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _transfer = require("../transfer.js");

var _options = require("./common/options.js");

var _statefulDecorator = require("./common/stateful-decorator.js");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var _default = {
  title: 'Transfer add & remove highlighted options',
  decorators: [(0, _statefulDecorator.statefulDecorator)()]
};
exports.default = _default;

const HasOptions = (_, {
  onChange,
  selected
}) => /*#__PURE__*/_react.default.createElement(_transfer.Transfer, {
  filterable: true,
  selected: selected,
  onChange: onChange,
  options: _options.options
});

exports.HasOptions = HasOptions;

const HasSelected = (_, {
  onChange,
  selected
}) => /*#__PURE__*/_react.default.createElement(_transfer.Transfer, {
  onChange: onChange,
  selected: selected,
  options: _options.options
});

exports.HasSelected = HasSelected;
HasSelected.story = {
  decorators: [(0, _statefulDecorator.statefulDecorator)({
    initialState: _options.options.slice(0, 4).map(({
      value
    }) => value)
  })]
};
