// @dhis2-ui/transfer/build/cjs/__e2e__/filter-options-list.stories.e2e.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.ControlledFilter = exports.AncCustomFilter = exports.LowercaseSearch = exports.UppercaseSearch = exports.SomeResults = exports.EmptyResult = exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _transfer = require("../transfer.js");

var _options = require("./common/options.js");

var _statefulDecorator = require("./common/stateful-decorator.js");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var _default = {
  title: 'Transfer filtering',
  decorators: [(0, _statefulDecorator.statefulDecorator)()]
};
exports.default = _default;

const EmptyResult = (_, {
  selected,
  onChange
}) => /*#__PURE__*/_react.default.createElement(_transfer.Transfer, {
  filterable: true,
  initialSearchTerm: "Foobarbaz",
  selected: selected,
  onChange: onChange,
  sourceEmptyPlaceholder: /*#__PURE__*/_react.default.createElement("span", {
    "data-test": "no-results"
  }, "No results"),
  options: _options.options
});

exports.EmptyResult = EmptyResult;

const SomeResults = (_, {
  selected,
  onChange
}) => /*#__PURE__*/_react.default.createElement(_transfer.Transfer, {
  selected: selected,
  filterable: true,
  initialSearchTerm: "ANC",
  onChange: onChange,
  options: _options.options
});

exports.SomeResults = SomeResults;

const UppercaseSearch = (_, {
  selected,
  onChange
}) => /*#__PURE__*/_react.default.createElement(_transfer.Transfer, {
  selected: selected,
  filterable: true,
  initialSearchTerm: "ANC",
  onChange: onChange,
  options: _options.options
});

exports.UppercaseSearch = UppercaseSearch;

const LowercaseSearch = (_, {
  selected,
  onChange
}) => /*#__PURE__*/_react.default.createElement(_transfer.Transfer, {
  filterable: true,
  initialSearchTerm: "anc",
  selected: selected,
  onChange: onChange,
  options: _options.options
});

exports.LowercaseSearch = LowercaseSearch;

const AncCustomFilter = (_, {
  selected,
  onChange
}) => /*#__PURE__*/_react.default.createElement(_transfer.Transfer, {
  filterable: true,
  selected: selected,
  onChange: onChange,
  filterCallback: (options, filter) => options.filter(({
    label
  }) => label.match(`(^| )ANC .*${filter}`)),
  options: _options.options
});

exports.AncCustomFilter = AncCustomFilter;

window.customFilterCallback = (options, filter) => {
  return options.filter(({
    label
  }) => label.indexOf(filter) !== -1);
};

window.Cypress && window.Cypress.cy.spy(window, 'customFilterCallback');

const ControlledFilter = (_, {
  filter,
  onChange,
  onFilterChange,
  selected
}) => /*#__PURE__*/_react.default.createElement(_transfer.Transfer, {
  selected: selected,
  onChange: onChange,
  filterable: true,
  filterCallback: window.customFilterCallback,
  searchTerm: filter,
  onFilterChange: onFilterChange,
  options: _options.options
});

exports.ControlledFilter = ControlledFilter;
ControlledFilter.story = {
  decorators: [(0, _statefulDecorator.statefulDecorator)()]
};
