// @dhis2-ui/transfer/build/cjs/__e2e__/notify_at_end_of_list.stories.e2e.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.PartialPickedList = exports.PartialSourceList = exports.FullPickedList = exports.FullSourceList = exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _transfer = require("../transfer.js");

var _options = require("./common/options.js");

var _statefulDecorator = require("./common/stateful-decorator.js");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var _default = {
  title: 'Transfer End Of List',
  decorators: [(0, _statefulDecorator.statefulDecorator)()]
};
exports.default = _default;
window.onEndReached = window.Cypress ? window.Cypress.cy.stub() : () => console.log('onEndReached');
window.onEndReachedPicked = window.Cypress ? window.Cypress.cy.stub() : () => console.log('onEndReachedPicked');

const FullSourceList = (_, {
  selected,
  onChange
}) => /*#__PURE__*/_react.default.createElement(_transfer.Transfer, {
  options: _options.options,
  selected: selected,
  onChange: onChange,
  onEndReached: window.onEndReached
});

exports.FullSourceList = FullSourceList;

const FullPickedList = (_, {
  selected,
  onChange
}) => /*#__PURE__*/_react.default.createElement(_transfer.Transfer, {
  options: _options.options,
  selected: selected,
  onChange: onChange,
  onEndReachedPicked: window.onEndReachedPicked
});

exports.FullPickedList = FullPickedList;
FullPickedList.story = {
  decorators: [(0, _statefulDecorator.statefulDecorator)({
    initialState: _options.options.map(({
      value
    }) => value)
  })]
};

const PartialSourceList = (_, {
  selected,
  onChange
}) => /*#__PURE__*/_react.default.createElement(_transfer.Transfer, {
  options: _options.options.slice(0, 4),
  selected: selected,
  onChange: onChange,
  onEndReached: window.onEndReached
});

exports.PartialSourceList = PartialSourceList;

const PartialPickedList = (_, {
  selected,
  onChange
}) => /*#__PURE__*/_react.default.createElement(_transfer.Transfer, {
  options: _options.options,
  selected: selected,
  onChange: onChange,
  onEndReachedPicked: window.onEndReachedPicked
});

exports.PartialPickedList = PartialPickedList;
PartialPickedList.story = {
  decorators: [(0, _statefulDecorator.statefulDecorator)({
    initialState: _options.options.slice(0, 4).map(({
      value
    }) => value)
  })]
};
