// @dhis2-ui/organisation-unit-tree/build/cjs/__e2e__/common.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.createDecoratorCustomDataProvider = exports.createDecoratorStatefulMultiSelection = exports.StatefulMultiSelectionWrapper = exports.dataProviderData = exports.getOrganisationUnitData = exports.delayResponse = exports.namespace = void 0;

var _appRuntime = require("@dhis2/app-runtime");

var _propTypes = _interopRequireDefault(require("prop-types"));

var _react = _interopRequireWildcard(require("react"));

function _getRequireWildcardCache(nodeInterop) { if (typeof WeakMap !== "function") return null; var cacheBabelInterop = new WeakMap(); var cacheNodeInterop = new WeakMap(); return (_getRequireWildcardCache = function (nodeInterop) { return nodeInterop ? cacheNodeInterop : cacheBabelInterop; })(nodeInterop); }

function _interopRequireWildcard(obj, nodeInterop) { if (!nodeInterop && obj && obj.__esModule) { return obj; } if (obj === null || typeof obj !== "object" && typeof obj !== "function") { return { default: obj }; } var cache = _getRequireWildcardCache(nodeInterop); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (key !== "default" && Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj.default = obj; if (cache) { cache.set(obj, newObj); } return newObj; }

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

/* eslint-disable react/display-name */
const namespace = 'OrganisationUnitTree';
exports.namespace = namespace;

const delayResponse = (delay, response) => () => new Promise(resolve => setTimeout(() => resolve(response), delay));

exports.delayResponse = delayResponse;

const getOrganisationUnitData = id => {
  let data;

  if (id === 'A0000000000') {
    data = {
      id: 'A0000000000',
      path: '/A0000000000',
      displayName: 'Org Unit 1',
      children: [{
        id: 'A0000000001',
        path: '/A0000000000/A0000000001',
        children: [{
          id: 'A0000000003'
        }, {
          id: 'A0000000004'
        }],
        displayName: 'Org Unit 2'
      }, {
        id: 'A0000000002',
        path: '/A0000000000/A0000000002',
        children: [],
        displayName: 'Org Unit 3'
      }, {
        id: 'A0000000006',
        path: '/A0000000000/A0000000006',
        children: [],
        displayName: 'Org Unit 7'
      }]
    };
  }

  if (id === 'A0000000001') {
    data = {
      id: 'A0000000001',
      path: '/A0000000000/A0000000001',
      displayName: 'Org Unit 2',
      children: [{
        id: 'A0000000003',
        path: '/A0000000000/A0000000001/A0000000003',
        children: [],
        displayName: 'Org Unit 4'
      }, {
        id: 'A0000000004',
        path: '/A0000000000/A0000000001/A0000000004',
        children: [],
        displayName: 'Org Unit 5'
      }]
    };
  }

  if (id === 'A0000000002') {
    data = {
      displayName: 'Org Unit 3',
      id: 'A0000000002',
      path: '/A0000000000/A0000000002',
      children: []
    };
  }

  if (id === 'A0000000003') {
    data = {
      displayName: 'Org Unit 4',
      id: 'A0000000003',
      path: '/A0000000000/A0000000001/A0000000003',
      children: [{
        id: 'A0000000007',
        path: '/A0000000000/A0000000001/A0000000003/A0000000007',
        children: [],
        displayName: 'Org Unit 8'
      }]
    };
  }

  if (id === 'A0000000004') {
    data = {
      displayName: 'Org Unit 5',
      id: 'A0000000004',
      path: '/A0000000000/A0000000001/A0000000004',
      children: []
    };
  }

  if (id === 'A0000000006') {
    data = {
      displayName: 'Org Unit 7',
      id: 'A0000000006',
      path: '/A0000000000/A0000000006',
      children: []
    };
  }

  if (id === 'A0000000007') {
    data = {
      displayName: 'Org Unit 8',
      id: 'A0000000007',
      path: '/A0000000000/A0000000001/A0000000003/A0000000007',
      children: []
    };
  }

  return data;
};

exports.getOrganisationUnitData = getOrganisationUnitData;
const dataProviderData = {
  organisationUnits: (_, {
    id
  }) => {
    const data = getOrganisationUnitData(id);

    if (!data) {
      return Promise.reject(new Error('404 - Org unit not found'));
    }

    return Promise.resolve(data);
  }
};
exports.dataProviderData = dataProviderData;

const StatefulMultiSelectionWrapper = ({
  children,
  onSelectionChange = () => null
}) => {
  const [selected, setSelected] = (0, _react.useState)([]);
  return children({
    selected,
    onChange: ({
      selected: newSelected
    }) => {
      setSelected(newSelected);
      onSelectionChange(newSelected);
    }
  });
};

exports.StatefulMultiSelectionWrapper = StatefulMultiSelectionWrapper;
StatefulMultiSelectionWrapper.propTypes = {
  children: _propTypes.default.func.isRequired,
  onSelectionChange: _propTypes.default.func
};

const createDecoratorStatefulMultiSelection = args => {
  return Story => /*#__PURE__*/_react.default.createElement(StatefulMultiSelectionWrapper, {
    onSelectionChange: args === null || args === void 0 ? void 0 : args.onSelectionChange
  }, ({
    selected,
    onChange
  }) => /*#__PURE__*/_react.default.createElement(Story, {
    selected: selected,
    onChange: onChange
  }));
};

exports.createDecoratorStatefulMultiSelection = createDecoratorStatefulMultiSelection;

const createDecoratorCustomDataProvider = args => {
  const data = (args === null || args === void 0 ? void 0 : args.data) || dataProviderData;
  return Story => {
    window.dataProviderData = data;
    return /*#__PURE__*/_react.default.createElement(_appRuntime.CustomDataProvider, {
      data: data
    }, /*#__PURE__*/_react.default.createElement(Story, null));
  };
};

exports.createDecoratorCustomDataProvider = createDecoratorCustomDataProvider;
