// @dhis2-ui/organisation-unit-tree/build/cjs/__e2e__/expanded.stories.e2e.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.WithRootMainAndRootSubOrgUnit = exports.InitiallyExpandedPaths = exports.NoInitiallyExpandedPaths = exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _index = require("../index.js");

var _common = require("./common.js");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var _default = {
  title: 'OrganisationUnitTree/expanded',
  decorators: [(0, _common.createDecoratorStatefulMultiSelection)(), (0, _common.createDecoratorCustomDataProvider)()]
};
exports.default = _default;

const NoInitiallyExpandedPaths = (_, {
  onChange,
  selected
}) => /*#__PURE__*/_react.default.createElement(_index.OrganisationUnitTree, {
  roots: "A0000000000",
  selected: selected,
  onChange: onChange
});

exports.NoInitiallyExpandedPaths = NoInitiallyExpandedPaths;

const InitiallyExpandedPaths = (_, {
  onChange,
  selected
}) => /*#__PURE__*/_react.default.createElement(_index.OrganisationUnitTree, {
  roots: "A0000000000",
  selected: selected,
  onChange: onChange,
  initiallyExpanded: ['/A0000000000']
});

exports.InitiallyExpandedPaths = InitiallyExpandedPaths;

const WithRootMainAndRootSubOrgUnit = (_, {
  onChange,
  selected
}) => /*#__PURE__*/_react.default.createElement(_index.OrganisationUnitTree, {
  roots: ['A0000000000', 'A0000000001'],
  selected: selected,
  onChange: onChange
});

exports.WithRootMainAndRootSubOrgUnit = WithRootMainAndRootSubOrgUnit;
