// @dhis2-ui/header-bar/build/es/profile/profile-header.js
import _JSXStyle from "styled-jsx/style";
import { Box } from '@dhis2-ui/box';
import { useConfig } from '@dhis2/app-runtime';
import propTypes from '@dhis2/prop-types';
import React from 'react';
import { joinPath } from '../join-path.js';
import i18n from '../locales/index.js';
import { UserIcon } from '../user-icon/user-icon.js';

const ProfileName = ({
  children
}) => /*#__PURE__*/React.createElement("div", {
  "data-test": "headerbar-profile-username",
  className: "jsx-2223023701"
}, children, /*#__PURE__*/React.createElement(_JSXStyle, {
  id: "2223023701"
}, ["div.jsx-2223023701{margin-bottom:3px;font-size:16px;line-height:19px;}"]));

ProfileName.propTypes = {
  children: propTypes.string
};

const ProfileEmail = ({
  children
}) => /*#__PURE__*/React.createElement("div", {
  "data-test": "headerbar-profile-user-email",
  className: "jsx-1072768994"
}, children, /*#__PURE__*/React.createElement(_JSXStyle, {
  id: "1072768994"
}, ["div.jsx-1072768994{margin-bottom:6px;font-size:14px;line-height:16px;}"]));

ProfileEmail.propTypes = {
  children: propTypes.string
};

const ProfileEdit = ({
  children
}) => {
  const {
    baseUrl
  } = useConfig();
  return /*#__PURE__*/React.createElement("a", {
    href: joinPath(baseUrl, 'dhis-web-user-profile/#/profile'),
    "data-test": "headerbar-profile-edit-profile-link",
    className: "jsx-233684196"
  }, children, /*#__PURE__*/React.createElement(_JSXStyle, {
    id: "233684196"
  }, ["a.jsx-233684196{color:rgba(0,0,0,0.87);font-size:12px;line-height:14px;-webkit-text-decoration:underline;text-decoration:underline;cursor:pointer;}"]));
};

ProfileEdit.propTypes = {
  children: propTypes.string
};

const ProfileDetails = ({
  name,
  email
}) => /*#__PURE__*/React.createElement("div", {
  className: "jsx-3814112749"
}, /*#__PURE__*/React.createElement(ProfileName, null, name), /*#__PURE__*/React.createElement(ProfileEmail, null, email), /*#__PURE__*/React.createElement(ProfileEdit, null, i18n.t('Edit profile')), /*#__PURE__*/React.createElement(_JSXStyle, {
  id: "3814112749"
}, ["div.jsx-3814112749{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-flex-direction:column;-ms-flex-direction:column;flex-direction:column;margin-left:20px;color:#000;font-size:14px;font-weight:400;}"]));

ProfileDetails.propTypes = {
  email: propTypes.string,
  name: propTypes.string
};
export const ProfileHeader = ({
  name,
  email,
  avatarUrl
}) => /*#__PURE__*/React.createElement("div", {
  className: "jsx-3625287538"
}, /*#__PURE__*/React.createElement(Box, {
  width: "48px",
  height: "48px"
}, /*#__PURE__*/React.createElement(UserIcon, {
  avatarUrl: avatarUrl,
  name: name,
  dataTest: "headerbar-profile-menu-icon"
})), /*#__PURE__*/React.createElement(ProfileDetails, {
  name: name,
  email: email
}), /*#__PURE__*/React.createElement(_JSXStyle, {
  id: "3625287538"
}, ["div.jsx-3625287538{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-flex-direction:row;-ms-flex-direction:row;flex-direction:row;margin-left:24px;padding-top:20px;}"]));
ProfileHeader.propTypes = {
  avatarUrl: propTypes.string,
  email: propTypes.string,
  name: propTypes.string
};
