// @dhis2-ui/transfer/build/cjs/__e2e__/transferring-items.stories.e2e.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.SomeSelected = exports.HasOptions = exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _transfer = require("../transfer.js");

var _options = require("./common/options.js");

var _statefulDecorator = require("./common/stateful-decorator.js");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var _default = {
  title: 'Transfer Transferring Items',
  decorators: [(0, _statefulDecorator.statefulDecorator)()]
};
exports.default = _default;
window.options = _options.options;

const HasOptions = (_, {
  selected,
  onChange
}) => /*#__PURE__*/_react.default.createElement(_transfer.Transfer, {
  selected: selected,
  onChange: onChange,
  options: _options.options
});

exports.HasOptions = HasOptions;

const SomeSelected = (_, {
  selected,
  onChange
}) => /*#__PURE__*/_react.default.createElement(_transfer.Transfer, {
  selected: selected,
  onChange: onChange,
  options: _options.options
});

exports.SomeSelected = SomeSelected;
SomeSelected.story = {
  decorators: [(0, _statefulDecorator.statefulDecorator)({
    initialState: _options.options.slice(0, 4).map(({
      value
    }) => value)
  })]
};
