// @dhis2-ui/header-bar/build/es/header-bar.stories.e2e.js
import { Provider } from '@dhis2/app-runtime';
import React from 'react';
import { HeaderBar } from './header-bar.js';
export default {
  title: 'HeaderBarTesting',
  component: HeaderBar
};
const config = {
  baseUrl: 'https://domain.tld/',
  apiVersion: ''
};
export const Default = () => /*#__PURE__*/React.createElement(Provider, {
  config: config
}, /*#__PURE__*/React.createElement(HeaderBar, {
  appName: "Example!"
}));
export const ShowOnlineStatus = () => /*#__PURE__*/React.createElement(Provider, {
  config: { ...config,
    headerbar: {
      showOnlineStatus: true
    }
  }
}, /*#__PURE__*/React.createElement(HeaderBar, {
  appName: "Example!"
}));
export const PWAEnabled = () => /*#__PURE__*/React.createElement(Provider, {
  config: { ...config,
    pwaEnabled: true
  }
}, /*#__PURE__*/React.createElement(HeaderBar, {
  appName: "Example!"
}));
export const WithLastOnline = () => /*#__PURE__*/React.createElement(Provider, {
  config: { ...config,
    headerbar: {
      showOnlineStatus: true,
      onlineStatusInfo: 'LAST_ONLINE'
    }
  }
}, /*#__PURE__*/React.createElement(HeaderBar, {
  appName: "Example!"
}));
